src/pynxtools_raman_multiformat/reader.py

```diff
@@ -129,18 +229,17 @@ def read_txt_file(self, filepath):
         # Transform: [[A, B], [C, D], [E, F]] into [[A, C, E], [B, D, F]]
         data = [list(item) for item in zip(*data)]
 
-        #transform linewise read data to colum style data
-        data = np.transpose(data)
-
         # assign column data with keys
         data_dict = {
-            "data/x_values": data[:, 0],
-            "data/y_values": data[:, 1]
+            "data/x_values": data[0],
+            "data/y_values": data[1]
         }
 
-        self.txt_data = data_dict
-        self.txt_header = header_dict
 
+        self.raman_data = data_dict

```

Comment:
You could think to move lines 184 through 238 to the ´witec_reader.py´ file as a function like
```python
def parse_txt_file(filepath):
    with open(filepath, "r") as file:
            lines = file.readlines()
    [...]
    data_dict = {
            "data/x_values": data[0],
            "data/y_values": data[1]
        }

    return data_dict
```
and then here in the `reader.py`, you only do 
```python
from pynxtools_raman_multiformat.witec.witec_reader import parse_txt_file post_process_witec

class RamanReaderMulti(MultiFormatReader):
    [...]
    def handle_txt_file(self, filepath):
        self.raman_data = parse_txt_data(filepath)
        self.post_process = post_process_witec.__get__(self, RamanReaderMulti)

        return {}
```

Reply:
this sounds good, did it as you adviced